pulp_container/app/access_policy.py

```diff
@@ -43,3 +43,27 @@ def has_namespace_obj_perms(self, request, view, action, permission):
         """
         namespace = view.get_object().namespace
         return request.user.has_perm(permission, namespace)
+
+
+class RepositoryVersionAccessPolicyFromDB(AccessPolicyFromDB):
+    """
+    Access policy for ContainerRepositoryVersionViewSet.
+    """
+
+    def has_delete_containerrepository_versions_perms(self, request, view, action):
+        """
+        Check if a user has perms to delete a repository version
+        """
+        repository = view.get_object().repository
+        sufficient_repo_perms = (
+            "container.delete_containerrepository",
+            "container.sync_containerrepository",
+            "container.modify_content_containerrepository",
+            "container.build_image_containerrepository",
+        )
+        # any permission will do
+        for perm in sufficient_repo_perms:
+            if request.user.has_perm(perm) or request.user.has_perm(perm, repository):
+                return True
```

Comment:
One option is to check all permissions here and be able to tweak them without changing a default policy statements in the db. The downside is that a user won't see these details if they try to read the default policy via API.

Reply:
What about we call this helper `has_modify_containerrepository_version_perms` (i.e. rather what it is, and not what we intend to use it for in the default policy). The policy would then read "You can delete RV, if you can add RV by any means."
And still i'd like to give the admin the tools (a `container.delete_containerrepository_version` permission) to be able to say "Just being able to add content does not mean you can delete any repo version."

Reply:
you have a point here - maybe this is not correct to enable a user to remove a repo version if he can sync the repo. Now i wonder whether we should allow this only if he can delete repo or maybe go with option 4

Reply:
I mostly see this function as a convenience wrapper, as the policy is able to reflect the same "has_any_of_those_permissions" logic.
That being said, i'm not opposed to have such convenience.
That it allows us to change the policies meaning without even touching it, i'm undecided whether that is nice or problematic. But I could be calmed by giving it a descriptive name.